lib/pf/UnifiedApi/Controller/Config.pm

```diff
@@ -327,6 +334,351 @@ sub sort_items {
+    }
+
+    if (defined $max) {
+        $extra->{max_length} = $max;
+    }
+
+    return ;
+}
+
+=head2 field_type
+
+Find the field type
+
+=cut
+
+sub field_type {
+    my ($self, $field, $no_array) = @_;
+    return pf::UnifiedApi::GenerateSpec::fieldType($field, $no_array);
+}
+
+=head2 field_is_required
+
+Check if the field is required
+
+=cut
+
+sub field_is_required {
+    my ($self, $field) = @_;
+    return  $field->required ? $self->json_true() : $self->json_false();
+}
+
+=head2 resource_options
+
+Create the resource options
+
+=cut
+
+sub resource_options {
+    my ($self) = @_;
+    my $form = $self->form($self->item);
+    my (%defaults, %placeholders, %allowed, %meta);
+    my %output = (
+        defaults => \%defaults,
+        placeholders => \%placeholders,
+        allowed => \%allowed,
+        meta => \%meta,
+    );
+    my $inheritedValues = $self->resourceInheritedValues;
+    my $defaultValues = $self->default_values;
+    for my $field ($form->fields) {
+        my $name = $field->name;
+        next if $self->isResourceFieldSkippable($field);
+        $defaults{$name} = $self->field_default($field, $defaultValues);
+        $placeholders{$name} = $self->field_resource_placeholder($field, $inheritedValues);
+        $allowed{$name} = $self->field_allowed($field);
+        $meta{$name} = $self->field_meta($field);
+    }
+
+    return $self->render(json => \%output);
+}
+
+=head2 isResourceFieldSkippable
+
+Check if a Resource Field is Skippable
+
+=cut
+
+sub isResourceFieldSkippable {
+    my ($self, $field) = @_;
+    return $field->name eq 'id';
+}
+
+=head2 resourceInheritedValues
+
+Get the resource inherited values
+
+=cut
+
+sub resourceInheritedValues {
+    my ($self) = @_;
+    my $values = $self->config_store->readFromImported($self->id, 'id');
+    if ($values) {
+        $values = $self->cleanup_item($values);
+    }
+
+    return $values;
+}
+
+=head2 field_default
+
+Get the default value of a field
+
+=cut
+
+sub field_default {
+    my ($self, $field, $inheritedValues) = @_;
+    return $field->default // $inheritedValues ? $inheritedValues->{$field->name} : undef;
+}
+
+=head2 default_values
+
+Get the default values from the config section
+
+=cut
+
+sub default_values {
+    my ($self) = @_;
+    my $cs = $self->config_store;
+    my $default_section = $cs->default_section;
+    my $defaultValues;
+    if ($default_section) {
+        $defaultValues = $self->cleanup_item($cs->read($default_section, 'id'));
+    }
+
+    return $defaultValues;
+}
+
+=head2 field_placeholder
+
+Get the placeholder for the field
+
+=cut
+
+sub field_placeholder {
+    my ($self, $field) = @_;
+    my $name = $field->name;
+    my $cs = $self->config_store;
+    my $default_section = $cs->default_section;
+    my $value;
+    if ($default_section) {
+        my $item = $self->cleanup_item($cs->read($default_section, 'id'));
+        $value = $item->{$name};
+    }
+
+    return $value // do {
```

Comment:
That isn't really in our usual practice.

To keep things consistent, I'd say we use an if/else like everywhere else in the code